src/games/strategy/engine/data/AllianceTracker.java

```diff
@@ -60,16 +55,15 @@ protected void addToAlliance(final PlayerID player, final String allianceName) {
    * Returns the PlayerID's that are members of the alliance
    * specified by the String allianceName
    *
-   * @param allianceName
-   *        Alliance name
+   * @param allianceName Alliance name
    * @return all the players in the given alliance
    */
-  public HashSet<PlayerID> getPlayersInAlliance(final String allianceName) {
-    final Iterator<PlayerID> keys = m_alliances.keySet().iterator();
+  public Set<PlayerID> getPlayersInAlliance(final String allianceName) {
+    final Iterator<PlayerID> keys = alliances.keySet().iterator();
     final HashSet<PlayerID> rVal = new HashSet<>();
     while (keys.hasNext()) {
       final PlayerID player = keys.next();
-      final Collection<String> alliances = m_alliances.get(player);
+      final Collection<String> alliances = this.alliances.get(player);
```

Comment:
The intermediate variable can be removed. The whole method can be rewritten as a 2 or 3 liner, rather than a 5-7 liner. But, to do that refactor, we are in territory where there should be a test put in place to ensure regressions do not happen. 

The problem is then that this code is not very well suited for testing. The requirement of a game data object in the constructor makes construction difficult, and the test much less focused, more of an integration test than a unit test. To enable that test we need to do other things, so I stopped here and put any future work in other PRs.